### PR TITLE
add postInstall script hook to install playwright/chromium

### PR DESCRIPTION
**Pull request by aruniverse**

### CODE REVIEW

File: common/changes/@itwin/oidc-signin-tool/playwright-install_2023-03-15-14-18.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/oidc-signin-tool",
+      "comment": "",
```
Comment (aruniverse) [inline common/changes/@itwin/oidc-signin-tool/playwright-install_2023-03-15-14-18.json:5]:
```suggestion
      "comment": "add postInstall script hook to install playwright/chromium",
```

File: common/config/azure-pipelines/integration-test.yaml [hunk 2]
```diff
@@ -81,9 +81,8 @@ jobs:
 
   - script: node common/scripts/install-run-rush.js install
     displayName: Rush Install
-
-  - script: npx playwright install
-    displayName: Install Playwright
+    env:
+      PLAYWRIGHT_BROWSERS_PATH: $(Build.BinariesDirectory)/pw-browsers
```
Comment (aruniverse) [inline common/config/azure-pipelines/integration-test.yaml:85]: do you think they'll need to set this env in the yaml files for the op tests?

File: packages/oidc-signin-tool/installPlaywright.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+const { exec } = require("child_process");
+
+const INSTALL_SCRIPT = "npx playwright install chromium";
+
+// const controller = new AbortController();
```
Comment (aruniverse) [inline packages/oidc-signin-tool/installPlaywright.js:10]: could you delete the commented out code?

File: packages/oidc-signin-tool/src/TestBrowserAuthorizationClient.ts [hunk 4]
```diff
@@ -512,6 +517,19 @@ export class TestBrowserAuthorizationClient implements AuthorizationClient {
...
     return page;
   }
+
+  private async cleanup(
+    page: Page,
+    signal: AbortSignal,
+    waitForCallbackUrl: Promise<Request | boolean>
+  ) {
+    if (signal.aborted)
+      await page.reload();
```
Comment (aruniverse) [inline packages/oidc-signin-tool/src/TestBrowserAuthorizationClient.ts:527]: if aborted why do we need to reload the pg?
shouldnt it just continue to cleanup as normal?

File: common/config/azure-pipelines/integration-test.yaml (hunk 2 above)
Comment (aruniverse) [reply common/config/azure-pipelines/integration-test.yaml:85]: cc @paulius-valiunas

File: packages/oidc-signin-tool/src/TestBrowserAuthorizationClient.ts (hunk 4 above)
Comment (ben-polinsky) [reply packages/oidc-signin-tool/src/TestBrowserAuthorizationClient.ts:527]: I deleted the comment I had... 
If we're on an error page then the waitForRequest callback waiting on the callback URL will not fire again. Without the page reload, the promise is unresolved at exit and playwright must try to resolve it. However, the page object is gone at this point (we've closed it) - it's not just the callback itself which is fired.

The reload is necessary to get the callback to fire once more and resolve the promise.

File: common/config/azure-pipelines/integration-test.yaml (hunk 2 above)
Comment (ben-polinsky) [reply common/config/azure-pipelines/integration-test.yaml:85]: I fear, yes...

Comment (ben-polinsky) [reply common/config/azure-pipelines/integration-test.yaml]: Removed all playwright variables and I believe we are good.

**Review by ben-polinsky — APPROVED** [review]

**Pull request merged**